File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        curr_sum = 0 
        max_sum = float('-inf')

        for i , ele in enumerate(nums) :
            curr_sum += ele 

            if curr_sum <= 0 :
                max_sum = max(max_sum,ele) 
                curr_sum = 0 
            else : 
                max_sum = max(max_sum , curr_sum) 

        return max_sum ;          

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum = nums[0] ; 
        int localSum = 0 ; 

        for ( int ele : nums ) {
            localSum += ele ; 
            sum = Math.max(localSum , sum ) ; 
            if ( localSum < 0 ) localSum = 0 ; 
        }

        return sum ; 
    }
}